cylc/uiserver/handlers.py

```diff
@@ -59,7 +59,11 @@ def initialize(self, path):
     def get(self):
         """Render the UI prototype."""
         index = os.path.join(self._static, "index.html")
-        self.write(open(index).read())
+        user = self.get_current_user()
+        protocol = self.request.protocol
+        host = self.request.host
```

Comment:
In Tornado, `host` includes hostname and port, e.g. `localhost:8000`. With the `protocol`, and the user's server path, we have everything the JS client needs :+1: